hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/workflow/diagnostic/DiagnosticManager.java

```diff
@@ -0,0 +1,187 @@
+package com.blackducksoftware.integration.hub.detect.workflow.diagnostic;
+
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.blackducksoftware.integration.hub.detect.bomtool.ExtractionId;
+import com.blackducksoftware.integration.hub.detect.configuration.DetectConfigWrapper;
+import com.blackducksoftware.integration.hub.detect.configuration.DetectProperty;
+import com.blackducksoftware.integration.hub.detect.workflow.bomtool.BomToolEvaluation;
+import com.blackducksoftware.integration.hub.detect.workflow.codelocation.DetectCodeLocation;
+
+public class DiagnosticManager {
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private final DetectConfigWrapper detectConfigWrapper;
+
+    private File outputDirectory;
+
+    private final DiagnosticReportManager diagnosticReportManager;
+    private final DiagnosticLogManager diagnosticLogManager;
+    private final DetectRunManager detectRunManager;
+    private final DiagnosticFileManager diagnosticFileManager;
+
+    private boolean isDiagnosticProtected = false;
+    private boolean isDiagnostic = false;
+
+    public DiagnosticManager(final DetectConfigWrapper detectConfigWrapper, final DiagnosticReportManager diagnosticReportManager, final DiagnosticLogManager diagnosticLogManager,
+            final DetectRunManager detectRunManager, final DiagnosticFileManager diagnosticFileManager) {
+        this.detectConfigWrapper = detectConfigWrapper;
+        this.diagnosticReportManager = diagnosticReportManager;
+        this.diagnosticLogManager = diagnosticLogManager;
+        this.detectRunManager = detectRunManager;
+        this.diagnosticFileManager = diagnosticFileManager;
+    }
+
+    public void init(final boolean isDiagnostic, final boolean isDiagnosticProtected) {
+
+        this.isDiagnostic = isDiagnostic;
+        this.isDiagnosticProtected = isDiagnosticProtected;
+
+        if (!isDiagnostic) {
+            return;
+        }
+
+        System.out.println("");
+        System.out.println("++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++");
+        System.out.println("Diagnostic mode on. Run id " + detectRunManager.getRunId());
+        System.out.println("++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++");
```

Comment:
Should we be using consistent text section separators across detect? I know we have some constants for this somewhere.